### PR TITLE
Added a Typo3 Scanner (closes #568)

### LINKED ISSUE #568: Add a typo3 security vulnerability scanner
### Is your feature request related to a problem

As a security analyst i would like to use the secureCodeBox to check my external attack surface. Especially CMS systems like Wordpress or Typo3 are common systems that may introduce new vulnerabilites on a regular basis. 

The BlogPost of @JavanXD https://javan.de/securing-typo3-cms-new-security-scanner/ also motivates this topic.

### Describe the solution you'd like

Since the secureCodeBox already supports the Wordpress scanner WPScan it would be great to also add at least one Typo3 scanner. There are two candidates (referring to the blog post):
- https://github.com/JavanXD/Typo3AccessChecker – Check if Typo3 security guidelines are followed.
- https://github.com/whoot/Typo3Scan – Typo3 Enumerator: Enumerates extensions to gain information about outdated and 

### Additional context

- https://javan.de/securing-typo3-cms-new-security-scanner/
- https://github.com/JavanXD/Typo3AccessChecker – Check if Typo3 security guidelines are followed.
- https://github.com/whoot/Typo3Scan – Typo3 Enumerator: Enumerates extensions to gain information about outdated and vulnerable extensions.

## Steps to implement a new scanner
Hint: A general guide how to implement a new SCB scanner is documented [here](https://docs.securecodebox.io/docs/contributing/integrating-a-scanner)

- [x] Create a new folder with the name of the [scanner here](https://github.com/secureCodeBox/secureCodeBox/tree/master/scanners)
- [x] Add a `README.gotmpl` and give a [brief overview](https://docs.securecodebox.io/docs/contributing/integrating-a-scanner/readme) of the scanner and its configuration options.
- [x] Add a HelmChart and document all configuration options.
- [x] Implement a new scanner specific `scan-type.yaml`
- [x] Implement a new scanner specific `parse-definition.yaml`
- [ ] Add (optional) some `cascading-rules.yaml` like documented [here](https://docs.securecodebox.io/docs/api/crds/cascading-rule)
- [x] Add (optional) a `Dockerfile` for the scanner if there is no existing one publicly available on dockerHub
- [x] Use the [parser-SDK](https://github.com/secureCodeBox/secureCodeBox/tree/master/parser-sdk) to implement a new findings parser (currently based on NodeJS)
- [x] Add unit tests with at minimum 80% test coverage
- [x] Add some example `scan.yaml` and `finding.yaml` files in the [example folder](https://docs.securecodebox.io/docs/contributing/integrating-a-scanner/examples-dir)
- [ ] Implement a [new integration or E2E test](https://docs.securecodebox.io/docs/contributing/integrating-a-scanner/integration-tests) for the hook [here](https://github.com/secureCodeBox/secureCodeBox/tree/master/tests/integration)

### PR DESCRIPTION
**Pull request by Ilyesbdlala**
## Description

Added a new Scannner that that scans websites using TYPO3 content management system: [typo3scan](https://github.com/whoot/Typo3Scan). It detects core vulnerabilities and scans for installed extensions. From these extensions, the vulnerable ones can be identified.

### Checklist

* [X] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [X] Make sure `npm test` runs for the whole project.
* [X] Make codeclimate checks happy

### CODE REVIEW

File: scanners/zap-advanced/.helmignore [hunk 1]
```diff
@@ -5,7 +5,13 @@
 .DS_Store
 
 parser/
-scanner/
+# this doesn't look too good but is required so that the scanners/scripts folder is included
```
Comment (SebieF) [inline scanners/zap-advanced/.helmignore:8]: Should this file be in the pull request?

**Review by SebieF — CHANGES_REQUESTED** [review]
Nicely done :) Possible additions would be integration tests and examples for cascading scans (could be combined with whatweb e.g.)

File: scanners/zap-advanced/.helmignore (hunk 1 above)
Comment (Ilyesbdlala) [reply scanners/zap-advanced/.helmignore]: No it shouldn't. I will remove it from the PR

File: scanners/typo3scan/scanner/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#Install git and Clone Repo 
+RUN apk add git \
+    && git clone https://github.com/whoot/Typo3Scan.git \
+    && cd Typo3Scan \
+    && rm -r .git .github doc
+COPY wrapper.sh /Typo3Scan/wrapper.sh
+
+#Runtime Image
+FROM python:3.9-alpine as runtime
```
Comment (SebieF) [inline scanners/typo3scan/scanner/Dockerfile:16]: Why are two different images necessary (base and runtime)? Seems like the runtime image just copies a directory from the base image and adds privileges and the result directory, should be doable in one image..

File: scanners/typo3scan/templates/typ3scan-parse-definition.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: execution.securecodebox.io/v1
+kind: ParseDefinition
+metadata:
+  name: typo3scan-json
+spec:
+  image: "{{ .Values.parser.image.repository }}:{{ .Values.parser.image.tag | default .Chart.Version }}"
+  imagePullSecrets:
+  - name: dockerhub-token
+  ttlSecondsAfterFinished: 60
```
Comment (SebieF) [inline scanners/typo3scan/templates/typ3scan-parse-definition.yaml:9]: The parse definition does not match with the standards from other scanners, e.g. gitleaks or nmap. Is there a reason for that?
```suggestion
spec:
  image: "{{ .Values.parser.image.repository }}:{{ .Values.parser.image.tag | default .Chart.Version }}"
  ttlSecondsAfterFinished: {{ .Values.parser.ttlSecondsAfterFinished }}
  env: 
    {{- toYaml .Values.parser.env | nindent 4 }}
```
(copied from nmap)

File: scanners/typo3scan/.helm-docs.gotmpl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+*/ -}}
+
+{{- define "extra.docsSection" -}}
+---
+title: "Typo3Scan"
+category: "scanner"
+type: "Network"
+state: "released"
+appVersion: "{{ template "chart.appVersion" . }}"
+usecase: "Automation of the process of detecting the Typo3 CMS, and it's installed extensions"
```
Comment (SebieF) [inline scanners/typo3scan/.helm-docs.gotmpl:14]: Typo:
```suggestion
usecase: "Automation of the process of detecting the Typo3 CMS and its installed extensions"
```

File: scanners/typo3scan/scanner/Dockerfile (hunk 2 above)
Comment (SebieF) [reply scanners/typo3scan/scanner/Dockerfile:16]: Okay I think that's a better way than just deleting unnecessary installations and directories :+1:
